docs/src/pages/[platform]/connected-components/storage/storagemanager/examples/StorageManagerEventExample.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { StorageManager } from '@aws-amplify/ui-react-storage';
+
+export const StorageManagerEventExample = () => {
+  // NOTE: you will need to keep a ref of the files because
+  // the StorageManager calls onUploadSuccess and onUploadError asynchronously
+  // when each file is uploaded.
+  const filesRef = React.useRef([]);
+  const [files, setFiles] = React.useState([]);
+
+  return (
+    <>
+      <StorageManager
+        acceptedFileTypes={['image/*']}
+        accessLevel="public"
+        maxFileCount={10}
+        onFileRemove={({ key }) => {
+          // remove the file from the list
+          filesRef.current = filesRef.current.filter(
+            (file) => file.key !== key
+          );
+          setFiles(filesRef.current);
+        }}
+        onUploadSuccess={({ key }) => {
+          filesRef.current = [...filesRef.current, { key }];
+          setFiles(filesRef.current);
+        }}
```

Comment:
Not fully grasping why we need the ref here, can these just be:
```suggestion
        onFileRemove={({ key }) => {
          setFiles((prevFiles) => prevFiles.filter(
            (file) => file.key !== key
          ));
        }}
        onUploadSuccess={({ key }) => {
          setFiles((prevFiles) => [...prevFiles, { key });
        }}
```